src/libraries/Microsoft.PowerFx.Core/Public/Types/FormulaType.cs

```diff
@@ -73,7 +89,7 @@ internal static FormulaType Build(DType type)
                     return new RecordType(DType.CreateRecord(type.GetAllNames(DPath.Root)));
 
                 default:
-                    throw new NotImplementedException($"Not implemented type: {type}");
```

Comment:
>  throw new NotImplementedException($"Not implemented type: {type}"); [](http://example.com/codeflow?start=18&length=69)

Can we keep the exception?
Introducing Invalid here can cover up real issues.

Reply:
I added a new function BuildOrInvalid that can return Invalid.  Existing calls to Build will still throw.